grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -457,12 +458,23 @@ class Product implements Comparable, Serializable {
     }
 
     /**
-     * Currently not implement since it would require coupling InventoryService to Product.
+     * Get quantity available to promise for this product given the location
      *
      * @param locationId
      */
-    def getQuantityAvailableToPromise(Integer locationId) {
-        throw new NotImplementedException()
+    def getQuantityAvailableToPromise(String locationId) {
+        def productAvailability = ProductAvailability.createCriteria().list {
```

Comment:
If you want to return only the quantityAvailableToPromise, then I think you could do `.get()` without a `resultTransformer` and with a simple `return productAvailability?:0`